File: inform/Settings/Specifics/IFBasicInformSettings.m
//
//  IFBasicInformSettings.m
//  Inform
//
//  Created by Toby Nelson on 25/05/2022.
//  Copyright 2022 Toby Nelson. All rights reserved.
//

#import "IFBasicInformSettings.h"
#import "IFUtility.h"
#import "IFCompilerSettings.h"

@implementation IFBasicInformSettings {
    IBOutlet NSButton* basicInform;
}

- (instancetype) init {
	return [self initWithNibName: @"BasicInformSettings"];
}

- (NSString*) title {
	return [IFUtility localizedString: @"Basic Inform"];
}

#pragma mark - Setting up 

- (void) updateFromCompilerSettings {
    IFCompilerSettings* settings = [self compilerSettings];
	
	[basicInform setState: [settings basicInform]?NSControlStateValueOn:NSControlStateValueOff];
}

- (void) setSettings {
    IFCompilerSettings* settings = [self compilerSettings];

	[settings setBasicInform: [basicInform state] == NSControlStateValueOn];
}

- (BOOL) enableForCompiler: (NSString*) compiler {
	// These settings only apply to Natural Inform
	if ([compiler isEqualToString: IFCompilerNaturalInform])
		return YES;
	else
		return NO;
}

@end

File: inform/Settings/Specifics/IFBasicInformSettings.swift
//
//  IFBasicInformSettings.swift
//  Inform
//
//  Created by Toby Nelson on 28/09/2023.
//

import Foundation

class IFBasicInformSettings : IFSetting {
    @IBOutlet var basicInform: NSButton!

    convenience init() {
        self.init(nibName:"BasicInformSettings")
    }

    override var title: String! {
        return IFUtility.localizedString("Basic Inform")
    }

    override func updateFromCompilerSettings() {
        let settings:IFCompilerSettings! = self.compilerSettings

        basicInform.state = settings.basicInform ? NSControl.StateValue.on : NSControl.StateValue.off
    }

    override func setSettings() {
        let settings:IFCompilerSettings! = self.compilerSettings

        settings.basicInform = basicInform.state == NSControl.StateValue.on
    }

    override func enable(forCompiler compiler: String!) -> Bool {
        // These settings only apply to Natural Inform
        return compiler == IFCompilerNaturalInform
    }
}
